src/scroll-timeline-base.js

```diff
@@ -184,47 +194,61 @@ function updateSource(timeline, source) {
       // Store a list of connected timelines and current scroll position
       details = {
         timelines: [],
-        scrollPos: {
+        measurements: {
           scrollLeft: source.scrollLeft,
-          scrollTop: source.scrollTop
+          scrollTop: source.scrollTop,
+          scrollWidth: source.scrollWidth,
+          scrollHeight: source.scrollHeight,
+          clientWidth: source.clientWidth,
+          clientHeight: source.clientHeight
         }
       };
       sourceDetails.set(source, details);
 
       const resizeObserver = new ResizeObserver(() => {
         // Sample and store scroll pos
-        details.scrollPos = {
+        details.measurements = {
           scrollLeft: source.scrollLeft,
-          scrollTop: source.scrollTop
+          scrollTop: source.scrollTop,
+          scrollWidth: source.scrollWidth,
+          scrollHeight: source.scrollHeight,
+          clientWidth: source.clientWidth,
+          clientHeight: source.clientHeight
         };
+        // Update dimensions of ranges before ticking the timelines
+        for (const timeline of details.timelines) {
+          updateRanges(timeline);
+        }
         requestAnimationFrame(() => {
           // Defer ticking timeline to animation frame to prevent
           // "ResizeObserver loop completed with undelivered notifications"
           for (const timeline of details.timelines) {
-            renormalizeAnimationTimings(timeline);
+            updateInternal(timeline);
           }
         });
       });
       resizeObserver.observe(source);
+      for (const child of source.children) {
+        resizeObserver.observe(child);
```

Comment:
It was necessary to pass some of the subtests, but we can definitely defer this.

Reply:
I have dropped the child observations. It doesn't cause any regressions, so that was definitely excessive 🙏

I have, however, added a mutation observer for class and style attributes to prevent regressions in /scroll-animations/scroll-timelines/scroll-animation-inactive-timeline.html.

I hope that's OK.